File: app/src/main/java/com/ibrahem/hamza/ebook/Actvities/SplashScreen.java
package com.ibrahem.hamza.ebook.Actvities;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import ibrahem.hamza.ebook.R;
import ibrahem.hamza.ebook.databinding.ActivitySplashBinding;


public class SplashScreen extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        ActivitySplashBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_splash);

        Animation animation = AnimationUtils.loadAnimation(this, R.anim.splashtransition);

        binding.SplashImage.startAnimation(animation);
        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashScreen.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }, 1250);

    }
}

File: app/src/main/java/com/ibrahem/islamic/ebook/app/ui/Activites/SplashScreen/SplashScreen.kt
package com.ibrahem.islamic.ebook.app.ui.Activites.SplashScreen

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import com.ibrahem.islamic.ebook.app.ui.Activites.mainActivity.MainActivity
import ibrahem.islamic.ebook.R
import ibrahem.islamic.ebook.databinding.ActivitySplashBinding

@SuppressLint("CustomSplashScreen")
class SplashScreen : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val binding = ActivitySplashBinding.inflate(layoutInflater)

        val animation = AnimationUtils.loadAnimation(this, R.anim.splashtransition)

        binding.SplashImage.startAnimation(animation)
        val handler = Handler()
        handler.postDelayed({
            val intent = Intent(this@SplashScreen, MainActivity::class.java)
            startActivity(intent)
            finish()

        }, 1200)
    }
}